mathlibtools/lib.py

```diff
@@ -522,16 +522,20 @@ def run(self, args: List[str]) -> str:
         """Run a command in the project directory, and returns stdout + stderr.
 
            args is a list as in subprocess.run"""
-        return subprocess.run(args, cwd=str(self.directory), 
+        if len(args):
+            args[0] = shutil.which(args[0])
+        return subprocess.run(args, cwd=str(self.directory),
                               stderr=subprocess.STDOUT,
                               stdout=subprocess.PIPE,
                               check=True).stdout.decode()
-    
+
     def run_echo(self, args: List[str]) -> None:
         """Run a command in the project directory, letting stdin and stdout
         flow.
 
            args is a list as in subprocess.run"""
+        if len(args):
```

Comment:
This test is useless (and it should simply be `if args:` if it were useful) because `subprocess.run` would raise an exception anyway.